types/json5/index.d.ts

```diff
@@ -40,5 +41,18 @@ Comments
 
 Both inline (single-line) and block (multi-line) comments are allowed.
   */
-declare var json5: JSON;
+
+type JSONReplacer = (key: string, value: any) => any | (number | string)[] | null;
+
+interface JSON5 {
+    // Old JSON methods
+    parse(text: string, reviver?: (key: any, value: any) => any): any;
```

Comment:
Rather than duplicating these methods which already exist on the `JSON` interface in the lib, you can extend the existing one with the following:
```
declare global {
  interface JSON {
    stringify(value: any, options?: { space?: number | string, quote?: string, replacer?: (key: string, value: any) => any }): string;
  }
}
```
This will just add in the overload to the existing interface.

Reply:
For details, see the bottom of [this page of the handbook](https://www.typescriptlang.org/docs/handbook/declaration-merging.html).

Reply:
Thanks. This piece was missing and I wasn't aware that you could simply merge interfaces.

Reply:
This is fine. By doing this, you're not changing the global JSON interface, but only extending it when used through the JSON5 package. So the following is true:
```
import JSON5 = require('json5');

JSON5.stringify({ foo: "bar" }, { space: 4 }); // works fine, exposes 3 overloads for stringify
JSON.stringify({ foo: "bar" }, { space: 4 }); // causes error
JSON.stringify({ foo: "bar" }, undefined, 4); // works fine, exposes 2 overloads for stringify
```